pkg/http_endpoint/v1/router.go

```diff
@@ -56,26 +60,645 @@ func authRouter(clients *common.ClientContainer,
 	return router
 }
 
-func adminRouter(authMiddleware func(http.Handler) http.Handler, secret string) *chi.Mux {
-	router := chi.NewRouter()
-	router.Use(authMiddleware)
-	router.Use(httpAuth.AdminAuthenticationMiddleware(secret))
-	/*
-		router.Get("/", func(w http.ResponseWriter, r *http.Request) {
-			_, err := w.Write([]byte("You are admin\n"))
-			if err != nil {
+// GetDatasource method gets the datasource list
+func GetDatasource(clients *common.ClientContainer) http.HandlerFunc {
+		return func(w http.ResponseWriter, r *http.Request) {
+	err := clients.Grafana.DoLogon()
+	if err != nil {
+		switch err.(type) {
+		// common.InvalidUsernamePassword means, that user provided invalid
+		// username/password. We have to return 401 error then
+		case common.InvalidUsernamePassword:
+			common.WriteErrorToResponse(w, http.StatusUnauthorized,
+				"Unauthorized", err.Error())
+			return
+		// If any other error happened -> return 500 error
+		default:
+			log.Logger.Error(err)
+			common.WriteErrorToResponse(w, http.StatusInternalServerError,
+				http.StatusText(http.StatusInternalServerError),
+				"Internal server error occured")
+			return
+		}
+	}
+	dslist, err := clients.Grafana.GetDataSourceList()
+	if err != nil {
+		log.Logger.Error(err)
+		common.WriteErrorToResponse(w, http.StatusInternalServerError,
+			http.StatusText(http.StatusInternalServerError),
+			"Internal server error occured")
+		return
+	}
+	datasources, _ := json.Marshal(dslist)
+	w.WriteHeader(http.StatusOK)
+	w.Write(datasources)
+		}
+}
+
+// GetDatasourceID method gets the datasource by ID
+func GetDatasourceID(clients *common.ClientContainer) http.HandlerFunc {
+                return func(w http.ResponseWriter, r *http.Request) {
+	err := clients.Grafana.DoLogon()
+        if err != nil {
+                switch err.(type) {
+                // common.InvalidUsernamePassword means, that user provided invalid
+                // username/password. We have to return 401 error then
+                case common.InvalidUsernamePassword:
```

Comment:
I don't think it is valid, because it is not User's problem, that our admin grafana creds are invalid.
also, you return err.Error() that can contain some creds.
I would recommend to return 500